unity/EditorPlugin/UnitTesting/UnityEditorTestLauncher.cs

```diff
@@ -58,9 +64,10 @@ public void TryLaunchUnitTests()
         string testEditorAssemblyProperties =  testEditorAssembly.GetType().GetProperties().Select(a=>a.Name).Aggregate((a, b)=>a+ ", "+b);
         string testEngineAssemblyProperties = testEngineAssembly.GetType().GetProperties().Select(a=>a.Name).Aggregate((a, b)=>a+ ", "+b);
 
-        var launcherType = testEditorAssembly.GetType(EditModeLauncher);
+        var launcherTypeString = myLaunch.TestMode == TestMode.Edit ? EditModeLauncher : PlayModeLauncher;
+        var launcherType = testEditorAssembly.GetType(launcherTypeString);
         if (launcherType == null)
-          throw new NullReferenceException($"Could not find {EditModeLauncher} among {testEditorAssemblyProperties}");
+          throw new NullReferenceException($"Could not find {launcherTypeString} among {testEditorAssemblyProperties}");
```

Comment:
Perhaps use `Assertion.AssertNotNull` here instead of throwing `NullReferenceException`? 

That would change behaviour, though - `AssertNotNull` doesn't throw in release mode. If we expect this to never be null, that's ok, but if we expect it to be null, perhaps we should return gracefully instead of throwing?